### DISCUSSION TITLE
How do I check if an existing cookie is httpOnly inside an API route?

### DISCUSSION THREAD

**Discussion by OZZlE** [vercel/next.js#58135] [Help]
### Needed knowlege for answers:
- API Routes: https://nextjs.org/docs/pages/building-your-application/routing/api-routes
- https://nextjs.org/docs/app/api-reference/functions/cookies
- https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts

### Summary

I tried https://nextjs.org/docs/app/api-reference/functions/cookies

    import { cookies } from 'next/headers';
    
    const cookieStore = cookies()
    const token = cookieStore.get('token');
    console.log('token ', token )

but it gives the error:
error - node_modules\next\dist\client\components\headers.js (34:14) @ cookies
error - unhandledRejection: Error: Invariant: Method expects to have requestAsyncStorage, none available

I think generally it would be great if the documentation could explain how to do things on Pages as well as inside API routes (if relevant) :) otherwise I love Next.js. Big thanks.

**Comment by luanbitar** [comment]
Hey @OZZlE, if you're trying to access this cookie on the client side, you need to put your code inside of the component, although, if you're trying to access it on the server side, you need to add `"use server"` at the top and move your code to inside of your function.
Can you provide a code example for your problem?

**Reply by OZZlE** [reply]
I'm not using the App Router so "use server" / "use client" does nothing in my experience. I like the old still supported setup with Pages directory, api etc.

I'm not sure how it's even possible to make this more clear... do you know what API Route means? Plz read up on https://nextjs.org/docs/pages/building-your-application/routing/api-routes

Sorry if I am being rude, I'm just crazy about being effective in my work..

**Reply by OZZlE** [reply]
Code you put inside an API Route never runs on the client

**Reply by luanbitar** [reply]
You should provide more information next time. You need to use the pages router [API for cookies](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#accessing-request-object), your solution should be using `req.cookies`
